src/sync/remote_mongo_collection.cpp

```diff
@@ -18,3 +18,472 @@
 
 #include "sync/remote_mongo_collection.hpp"
 
+namespace realm {
+namespace app {
+
+static void handle_response(util::Optional<AppError> error,
+                            util::Optional<std::string> value,
+                            std::function<void(std::string, util::Optional<AppError>)> completion_block)
+{
+    if (value && !error) {
+        return completion_block(*value, error);
+    }
+    
+    return completion_block("", error);
+}
+
+static void handle_response_optional(util::Optional<AppError> error,
+                            util::Optional<std::string> value,
+                            std::function<void(Optional<std::string>, util::Optional<AppError>)> completion_block)
```

Comment:
something seems off that we have these two versions, smells like some API inconsistency if we have completion blocks taking an optional string and a non-optional string (but it can be empty) - can we settle on one version for consistency?

Reply:
I agree, all strings will have an optional type, so I've moved all calls to be consistent with that